### DISCUSSION TITLE
Which part of TSDX displays "Circular dependency" warnings in builds?

### DISCUSSION THREAD

**Discussion by thehappybug** [jaredpalmer/tsdx#906] [Q&A] [answered]
Hey there. I'm working on refactoring a large project and unfortunately, there are a couple of files with circular dependencies.

I've done my homework on why this is bad and why I need to remove this eventually. But for now, I've ordered the initialization, import, and export parts to make the circularly dependent files work correctly.

However, for understanding why a correctly working code still generates build warnings, can you point out where (TypeScript/Babel/Rollup/etc) these "Circular Dependency" warnings are being created in TSDX?

![image](https://user-images.githubusercontent.com/3393530/96194917-d636e480-0f64-11eb-96db-7712ed8ed70c.png)

I am using TSDX version 0.14.0. Please find the complete devDependencies list below:

```json
{
    "@size-limit/preset-small-lib": "^4.6.0",
    "@types/lodash.flatten": "^4.4.6",
    "husky": "^4.3.0",
    "size-limit": "^4.6.0",
    "tsdx": "^0.14.0",
    "tslib": "^2.0.1",
    "typescript": "^3.9.3"
}
```

**Comment by agilgur5** [comment, marked as answer]
Good question, off the top my head I'd guess that to be Rollup as it's usually the one giving warnings like that.

Looked that up and indeed [looks like Rollup](https://github.com/rollup/rollup/blob/22289f9cdb63169dc914d3c6e0bb8cc4de99c6cc/src/Graph.ts#L201).

Aside: also looks like GitHub Discussions doesn't support templates... the issue templates are quite useful.

**Reply by thehappybug** [reply]
Thank you. I've turned off circular dep. warnings as they didn't make sense for my code.

Generating TS code is tough! 😁
